addons/godot-xr-tools/hands/collision_hand.gd

```diff
@@ -229,3 +301,57 @@ func _update_target() -> void:
 	# Use first target override if specified
 	if _target_overrides.size():
 		_target = _target_overrides[0].target
+
+
+# If a skeleton is set, update.
+func _update_hand_skeleton():
+	if hand_skeleton:
+		if hand_skeleton.has_signal("skeleton_updated"):
+			# Godot 4.3+
+			hand_skeleton.skeleton_updated.connect(_on_skeleton_updated)
+		else:
+			hand_skeleton.pose_updated.connect(_on_skeleton_updated)
+
+		# Run atleast once to init
+		_on_skeleton_updated()
+
+
+# Update our finger digits when our skeleton updates
+func _on_skeleton_updated():
+	if not hand_skeleton:
+		return
+
+	var bone_count = hand_skeleton.get_bone_count()
+	for i in bone_count:
+		var collision_node : CollisionShape3D
+		var offset : Transform3D
+		offset.origin = Vector3(0.0, 0.015, 0.0) # move to side of joint
+
+		var bone_name = hand_skeleton.get_bone_name(i)
+		if bone_name == "Palm_L":
+			offset.origin = Vector3(-0.02, 0.025, 0.0) # move to side of joint
+			collision_node = _palm_collision_shape
+		elif bone_name == "Palm_R":
+			offset.origin = Vector3(0.02, 0.025, 0.0) # move to side of joint
+			collision_node = _palm_collision_shape
+		elif bone_name.contains("Proximal") or bone_name.contains("Intermediate") or \
+			bone_name.contains("Distal"):
+			if _digit_collision_shapes.has(bone_name):
+				collision_node = _digit_collision_shapes[bone_name]
+			else:
+				collision_node = CollisionShape3D.new()
+				collision_node.name = bone_name
+				collision_node.shape = \
+					preload("res://addons/godot-xr-tools/hands/scenes/collision/hand_digit.shape")
+				add_child(collision_node, false, Node.INTERNAL_MODE_BACK)
+				_digit_collision_shapes[bone_name] = collision_node
+
+		if collision_node:
+			# TODO it would require a far more complex approach,
```

Comment:
This is something for a future PR. Also this logic would likely be more suited to be a `SkeletonModifier3D` implementation so that the skeleton is also limited in movement, something even more powerful when combined with full hand tracking.

That might be an XRT2 thing though.